modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java

```diff
@@ -267,27 +300,71 @@ public void setAgentSetting(String key, String value) {
             "set-cookie"
         ),
         OperatorDynamic,
+        NodeScope,
+        DeprecatedWarning
```

Comment:
by marking a setting with DeprecatedWarning we are indicating that this will be removed. This is simialar to my comment on the affix settings. https://github.com/elastic/elasticsearch/pull/104345#discussion_r1452076987
as per @stu-elastic 's comment it would have to be consulted with Breaking changes committee

I think it is worth rising now. We don't necessarily need to plan to remove it in near future, but we want to suggest users not to use it (with just a warning - as you have used).

Reply:
Is there any docs on the deprecation process @pgomulka? I'm not sure what needs to be done next.

Reply:
@pgomulka I've removed all deprecation warnings here. I'm going to move those into an independent PR to wrap up this part.

Reply:
sounds good. feel free to merge